src/wallet/wallet.cpp

```diff
@@ -561,11 +566,11 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
         Unlock(strWalletPassphrase);
 
         // if we are using HD, replace the HD seed with a new one
-        if (m_spk_man->IsHDEnabled()) {
-            m_spk_man->SetHDSeed(m_spk_man->GenerateNewSeed());
+        if (auto spk_man = m_spk_man.get()) {
+            if (spk_man->IsHDEnabled()) {
+                spk_man->SetupGeneration(true);
```

Comment:
In commit "Refactor: Move SetupGeneration code out of CWallet" (ef97f0b5451e4959c0d0bf649a71eec9b226210e)

No need to change now, but it seems like it'd probably be a good idea to check for an error from SetupGeneration here as well.